src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -62,6 +64,287 @@ static ShardPlacement * TupleToShardPlacement(TupleDesc tupleDesc,
 /* exports for SQL callable functions */
 PG_FUNCTION_INFO_V1(master_stage_shard_row);
 PG_FUNCTION_INFO_V1(master_stage_shard_placement_row);
+PG_FUNCTION_INFO_V1(citus_table_size);
+PG_FUNCTION_INFO_V1(citus_total_relation_size);
+PG_FUNCTION_INFO_V1(citus_relation_size);
+
+/*
+ * GetShardIntervalsOnNode takes WorkerNode then compare it with
+ * each placement of table. Returns intervals of table on that node as a list of intervals.
+ * Note that, same interval of a given shard exists on each worker node for reference tables.
+ */
+List *
+GetShardIntervalsOnNode(WorkerNode *workerNode, Oid distributedRelationId)
+{
+	DistTableCacheEntry *distTableCacheEntry = DistributedTableCacheEntry(
+		distributedRelationId);
+	char *workerNodeName = workerNode->workerName;
+	uint32 workerNodePort = workerNode->workerPort;
+	List *intervalList = NIL;
+	int shardIndex = 0;
+
+	/* if the table is reference table, directly returns an interval to it */
+	if (distTableCacheEntry->partitionMethod == DISTRIBUTE_BY_NONE)
+	{
+		ShardInterval *shardInterval = distTableCacheEntry->sortedShardIntervalArray[0];
+		intervalList = lappend(intervalList, shardInterval);
+		return intervalList;
+	}
+
+	/* note that this part only works on tables with streaming replication */
+	for (shardIndex = 0; shardIndex < distTableCacheEntry->shardIntervalArrayLength;
+		 shardIndex++)
+	{
+		ShardPlacement *placement =
+			distTableCacheEntry->arrayOfPlacementArrays[shardIndex];
+		char *shardNodeName = placement->nodeName;
+		uint32 shardNodePort = placement->nodePort;
+
+		if ((strcmp(shardNodeName, workerNodeName) == 0 && shardNodePort ==
+			 workerNodePort))
+		{
+			ShardInterval *shardInterval =
+				distTableCacheEntry->sortedShardIntervalArray[shardIndex];
+			intervalList = lappend(intervalList, shardInterval);
+		}
+	}
+
+	return intervalList;
+}
+
+
+/*
+ * GenerateSizeQueryOnMultiplePlacements generates a select size query to get
+ * size of multiple tables from the relation with distributedRelationId. Note that,
+ * different size functions supported by PG are also supported by this function. Check
+ * CitusSizeFunctionTypes for the list of supported size functions on Citus.
+ */
+StringInfo
+GenerateSizeQueryOnMultiplePlacements(Oid distributedRelationId, List *intervalsOnNode,
+									  CitusSizeFunctionTypes sizeFunctionType)
```

Comment:
Instead of CitusSizeFunctionTypes, we can pass the related pg_\*_size function as const char* here. So that we can completely remove CitusSizeFunctionTypes type.